Alert utility example

russellnelsonbriggs opened the pull request:
### Description
Provides an example module (alert_util.py) that provides a CLI to summarize Alerts using the SDK

### Issues Resolved
[List any issues this PR will resolve. e.g., Fixes #01]

### Check List
- [NA - example only ] New functionality includes testing.
  - [NA - example only ] All tests pass for Python 2.7+ & 3.4+(`$ tox`).
- [NA ] New functionality has been documented in the README if applicable.
  - [NA ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [NA ] New endpoints supported are updated in the endpoints-support.md file.
- [ NA] Changes are documented in the CHANGELOG.

russellnelsonbriggs committed:
Create alert_util.py

adding example module for counting\exporting alerts

russellnelsonbriggs committed:
Update alert_util.py

russellnelsonbriggs commented:
Hi folks, my build failed in Travis with the following that I am not sure what to do about...
Can anyone help me with this?  thank you in advance!

$ pyenv global system 3.6
pyenv: version `3.6' not installed
The command "pyenv global system 3.6" failed and exited with 1 during .

russellnelsonbriggs commented:
found this.  https://travis-ci.community/t/error-pyenv-version-3-6-not-installed-since-a-week/1400

any chance someone with access to the build env could check into it?  thank you!

russellnelsonbriggs committed:
Update alert_util.py

sijeesh commented:
@russellnelsonbriggs, Thank you for your contribution.
Let me try to fix this build issue.

sijeesh commented:
@russellnelsonbriggs, This issue has been fixed in PR #448 .
Now you can update your branch with the latest changes and push it.

russellnelsonbriggs commented:
Thank you sir!

russellnelsonbriggs committed:
Update alert_util.py

added better documentation, general command cleanup

russellnelsonbriggs committed:
Merge branch 'alert_example' of https://github.com/russellnelsonbriggs/python-hpOneView into alert_example

russellnelsonbriggs committed:
Update alert_util.py

fix for tox error
 examples/alert_util.py:106:20: E999 SyntaxError: invalid syntax

sijeesh commented on examples/alert_util.py:
```diff
@@ -0,0 +1,179 @@
+
+    def get_alerts(self, _filter='', _view=''):
+        print("\nFilter: {}\n".format(_filter))
+        resp = self.client.alerts.get('/rest/alerts?filter={}'.format(_filter))
```
Do we need two get calls in this method? 
Can we use one get_all method with the filter to get all the alerts then print the total count and return the result?

russellnelsonbriggs replied:
In this case yes.  The reasoning is:  if you run the get_all() on a large number of alerts (I was testing with 60k), the default page size is only 25!  This causes the get_all to be very, very slow.  So I found that if I give it a count value that is the largest possible (total), it then set the page size to be 1000, which returned my 60k alerts in under a minute.

AND, the only way to get the total number of alerts quickly, without actually trying to fetch them all, was to use the  get().  This way it only gets one page.

sijeesh replied:
Ok, got it.

sijeesh reviewed (approved):
LGTM
